modules/navier_stokes/src/physics/WCNSFVFluidHeatTransferPhysicsBase.C

```diff
@@ -0,0 +1,254 @@
+      have_scalar = true;
+    // Now we determine what kind of functor we are dealing with
+    else
+    {
+      if (getProblem().hasFunctorWithType<ADReal>(_thermal_conductivity_name[i],
+                                                  /*thread_id=*/0))
+        have_scalar = true;
+      else
+      {
+        if (getProblem().hasFunctorWithType<ADRealVectorValue>(_thermal_conductivity_name[i],
+                                                               /*thread_id=*/0))
+          have_vector = true;
+        else
+          paramError("thermal_conductivity",
+                     "We only allow functor of type ADReal or ADRealVectorValue for thermal "
+                     "conductivity! Functor '" +
+                         _thermal_conductivity_name[i] + "' is not of the requested type.");
+      }
+    }
+  }
+
+  if (have_vector && !_porous_medium_treatment)
+    paramError("thermal_conductivity", "Cannot use anisotropic diffusion with non-porous flows!");
+
+  if (have_vector == have_scalar)
+    paramError("thermal_conductivity",
+               "The entries on thermal conductivity shall either be scalars of vectors, mixing "
+               "them is not supported!");
+  return have_vector;
+}
+
+void
+WCNSFVFluidHeatTransferPhysicsBase::addInitialConditions()
+{
+  // For compatibility with Modules/NavierStokesFV syntax
+  if (!_has_energy_equation)
+    return;
+  if (!_define_variables && parameters().isParamSetByUser("initial_temperature"))
+    paramError(
+        "initial_temperature",
+        "T_fluid is defined externally of WCNSFVFluidHeatTransferPhysicsBase, so should the inital "
+        "condition");
+  // do not set initial conditions if we load from file
+  if (getParam<bool>("initialize_variables_from_mesh_file"))
+    return;
+  // do not set initial conditions if we are not defining variables
+  if (!_define_variables)
+    return;
+
+  InputParameters params = getFactory().getValidParams("FunctionIC");
+  assignBlocks(params, _blocks);
+
+  if (!_app.isRestarting() || parameters().isParamSetByUser("initial_temperature"))
+  {
+    params.set<VariableName>("variable") = _fluid_temperature_name;
+    params.set<FunctionName>("function") = getParam<FunctionName>("initial_temperature");
+
+    getProblem().addInitialCondition("FunctionIC", _fluid_temperature_name + "_ic", params);
+  }
+}
+
+void
+WCNSFVFluidHeatTransferPhysicsBase::addMaterials()
+{
+  // For compatibility with Modules/NavierStokesFV syntax
+  if (!_has_energy_equation)
+    return;
+
+  InputParameters params = getFactory().getValidParams("INSFVEnthalpyFunctorMaterial");
+  assignBlocks(params, _blocks);
+
+  params.set<MooseFunctorName>(NS::density) = _density_name;
+  params.set<MooseFunctorName>(NS::cp) = _specific_heat_name;
+  params.set<MooseFunctorName>("temperature") = _fluid_temperature_name;
+
+  getProblem().addMaterial(
+      "INSFVEnthalpyFunctorMaterial", prefix() + "ins_enthalpy_material", params);
```

Comment:
I think we can put in Ramiro's functor material here once it's merged. We can just add a boolean "solve_for_enthalpy" to make the change of variable too. 
https://github.com/idaholab/moose/pull/29532/files#diff-230652c10695051399dccb2366c0e4d1b6ba10acd6322d9c3e06b733f5608edf

@grmnptr can you please push this PR forward?

Reply:
Yeah I reviewed it yesterday. I'll ask if they can address the comments or if I should do it.

Reply:
I meant this PR as in this one. I think the follow up post Ramiro's can be done in a different PR